src/models/test/ThemeProvider.test.js

```diff
@@ -2,117 +2,93 @@
 /* eslint-disable react/no-multi-comp */
 import React from 'react'
 import { shallow, render, mount } from 'enzyme'
-import ThemeProvider, { CHANNEL_NEXT, CONTEXT_CHANNEL_SHAPE  } from '../ThemeProvider'
+import ThemeProvider, { ThemeConsumer } from '../ThemeProvider'
```

Comment:
Maybe instead of `ThemeConsumer` we should test this with the `withTheme` HOC and normal `styled` constructors instead. That way we are ensuring the external API doesn't change.

Reply:
Okay so would this be OK or do we want to use `expectCSSMatches` function as it's done in `theme.test.js`?
```js
  it('should merge its theme with an outer theme', () => {
    const outerTheme = { main: 'black' }
    const innerTheme = { secondary: 'black' }

    const childrenSpy = jest.fn()
    const MyDiv = ({ theme }) => {
      childrenSpy(theme)
      return null
    }
    const MyDivWithTheme = withTheme(MyDiv)

    render(
      <ThemeProvider theme={outerTheme}>
        <ThemeProvider theme={innerTheme}>
          <MyDivWithTheme />
        </ThemeProvider>
      </ThemeProvider>
    )

    expect(childrenSpy).toHaveBeenCalledTimes(1)
    expect(childrenSpy).toHaveBeenCalledWith({
      ...outerTheme,
      ...innerTheme,
    })
  })
```